File: src/chipschallenge/BlockMove.java

package chipschallenge;

import chipschallenge.Move.Moves;


public class BlockMove {

    public final Block block;
    public final Moves move;

    public BlockMove(Block b, Moves m) {
        block = b;
        move = m;
    }

    @Override
    public BlockMove clone() {
        return new BlockMove(block, move);
    }

}

File: ChipsChallenge.Shared/BlockMove.cs
namespace ChipsChallenge.Shared
{
    using Moves = Move.Moves;

    public class BlockMove
    {
        public readonly Block Block;
        public readonly Moves Move;

        public BlockMove(Block b, Moves m)
        {
            Block = b;
            Move = m;
        }

        public BlockMove Clone()
        {
            return new BlockMove(Block, Move);
        }
    }
}